utilities_common/util_base.py

```diff
@@ -19,6 +21,17 @@ class UtilHelper(object):
     def __init__(self):
         pass
 
+    def load_plugins(self, plugins_namespace):
+        """ Discover and load CLI plugins. Yield a plugin module. """
+
+        def iter_namespace(ns_pkg):
+            return pkgutil.iter_modules(ns_pkg.__path__, ns_pkg.__name__ + ".")
+
+        for _, module_name, ispkg in iter_namespace(plugins_namespace):
+            if ispkg:
+                continue
+            yield importlib.import_module(module_name)
```

Comment:
Would it be necessary to support an "order" for loading of plugins ? Or is it the expectation for the plugin to be self contained ?

Reply:
This might be needed if plugins have cross-dependencies which is against the classical plugin system design where plugins have no cross-dependencies, they only depend on the core.
App.Ext developer is forced to not depend on other plugins here. However, we do not want the infrastructure to put restrictions about CLI design. This shows that this plugin mechanism is not an ideal choice for CLI extensions here.
Again, CLI auto-generation based on YANG modules can help a lot here, since YANG will show cross-dependencies between modules and CLI can be regenerated from scratch on installation/uninstallation/upgrade.

Reply:
"But I don't see any model here for a CLI plugin to see what (other cli plugins/infra) is currently available"

In your plugin you can do this to list what plugins are available:
```
pkgutil.iter_modules(show.plugins.__path__, show.plugins.__name__)
```